app.rb

```diff
@@ -0,0 +1,98 @@
+
+module Enumerable
+  def my_each(&prc)
+    length.times do |i|
+      prc.call(self[i])
+    end
+    self
+  end
+
+  def my_each_with_index(&prc)
+    length.times do |i|
+      prc.call(self[i], i)
+    end
+    self
+  end
+  
+  def my_select(&prc)
+    selected = []
+    my_each do |el|
+      selected << el if prc.call(el) == true
+    end
+    selected
+  end
+  
+  def my_all?(arg = nil)
+    if arg
+      my_each { |ele| return false unless arg === ele }
+    elsif block_given?
+      my_each { |ele| return false unless yield(ele) }
+    else
+      my_each { |ele| return false unless ele }
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
+    if arg
+      my_each { |ele| return true unless arg === ele }
+    elsif block_given?
+      my_each { |ele| return true unless yield(ele) }
+    else
+      my_each { |ele| return true unless ele }
+    end
+    false
+  end
+  
+  def my_none?(arg = nil)
+    if arg
+      my_each { |ele| return false if arg === ele }
+    elsif block_given?
+      my_each { |ele| return false if yield(ele) }
+    else
+      my_each { |ele| return false if ele }
+    end
+    true
+  end
```

Comment:
- [ ] `my_none?` is not working as the original `none?` method, also is not passing all test cases. 

### `none?` is an Enumerable that has these characteristics:
- Returns `true` if all elements are false.
```
[nil, nil, nil].none?
# true
```
- Returns `false` if all elements in the return true
```
[1, 'string', true].none?
# false
```

[Enumerable#none?](https://ruby-doc.org/core-2.7.2/Enumerable.html#method-i-none-3F)